### DISCUSSION TITLE
Utility for bundling macOS apps containing Qt frameworks

### DISCUSSION THREAD

**Discussion by SCG82** [mapeditor/tiled#2954] [General]
https://github.com/SCG82/macdylibbundler is a fork of https://github.com/auriamg/macdylibbundler that finds, copies, and fixes up (install names, rpaths) all dependencies of a macOS app bundle, including frameworks & Qt plugins. My motivation was to avoid having to manually fix up app bundles or deal with running `macdeployqt` followed by some recursive find+otool to find what it missed.

usage:
```
dylibbundler -cd -of -f -a ./SomeApp.app -x ./SomeApp.app/Contents/Plugins/BinaryFile.so
```
It's used by OBS Studio (https://github.com/obsproject/obs-studio/blob/master/CI/full-build-macos.sh)

**Comment by bjorn** [comment]
Hey @SCG82, thanks for offering to help ease macOS distribution with your tool!

There are a few things I'm wondering about though:

* Why did you fork macdylibbundler? I see you've contributed some patches, so I guess you needed changes the original author didn't want?

* Did you consider looking into fixing up `macdeployqt` instead?

* What advantage does your fork of macdylibbundler have over the relatively small Ruby script I've been using for many years now? https://github.com/mapeditor/tiled/blob/master/dist/macos/fixup-install-names.rb

**Reply by SCG82** [reply]
Hey @bjorn, thanks for your response. There are two main reasons why my changes haven't been incorporated into https://github.com/auriamg/macdylibbundler:

1. The changes I've made to the original macdylibbundler are too extensive (and some are breaking) to be submitted upstream in a single PR. I haven't had the time to break it apart into multiple PRs for @auriamg to review.
2. These changes haven't been extensively tested, besides myself https://github.com/obsproject/obs-studio.git is the only other project that has used my fork (and both of us use it to build the same software: obs-studio).

I haven't looked into what needs to be done for macdeployqt to be a complete tool for packaging mac apps.

The advantages my fork has over your ruby script is:

- not needing to run macdeployqt prior to running your script
- will work even if one of the Qt dependencies adds another Qt dependency at some point
- isn't fixed to Qt 5
